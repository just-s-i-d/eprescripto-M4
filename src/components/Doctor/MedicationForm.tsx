import { Input, Select, Form } from "antd";
import { CloseOutlined } from "@ant-design/icons";
import { MedicationType } from "@constants/types";

const { Item } = Form;
type MedicationFormPropsType = {
  fields: MedicationType;
  removeable?: boolean;
  removeMedicationInput: (id: number) => void;
};
const MedicationForm = ({
  fields,
  removeMedicationInput,
  removeable,
}: MedicationFormPropsType) => {
  return (
    <>
      <div className="mt-4 mb-2 flex justify-between">
        <h2 className="text-xl">Medication</h2>
        {removeable && (
          <CloseOutlined
            onClick={() => removeMedicationInput(fields.id)}
            data-testid="remove-field"
            className="text-[16px]"
          />
        )}
      </div>

      <div className="w-full grid sm:grid-cols-2 md:grid-cols-4 gap-x-5 gap-y-3">
        <Item
          name={fields.medicine}
          label="Medicine Name"
          className="w-full"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
            {
              pattern: /^[a-zA-Z0-9.-]+$/,
              message: "Special characters not allowed",
            },
          ]}
        >
          <Input />
        </Item>
        <Item
          name={fields.dosage}
          label="Dosage"
          className="w-full"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
            {
              pattern: /^[0-9/mg]/,
              message: "Enter a valid dosage",
            },
          ]}
        >
          <Input />
        </Item>
        <Item
          name={fields.timesPerDay}
          label="Frequency"
          className="w-full"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
          ]}
        >
          <Select
            options={[
              {
                value: 1,
                label: "1/Day",
              },
              {
                value: 2,
                label: "2/Day",
              },
              {
                value: 3,
                label: "3/Day",
              },
            ]}
          />
        </Item>
        <div className="flex gap-6">
          <Item
            name={fields.instruction}
            className="w-full"
            label="Instruction"
            required
            rules={[
              {
                required: true,
                message: "Field cannot be empty",
              },
            ]}
          >
            <Select
              options={[
                {
                  value: "beforeFood",
                  label: "Before eating food",
                },
                {
                  value: "afterFood",
                  label: "After eating Food",
                },
              ]}
            />
          </Item>
        </div>
      </div>
    </>
  );
};
export default MedicationForm;
