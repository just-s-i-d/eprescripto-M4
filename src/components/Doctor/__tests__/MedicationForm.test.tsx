import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import MedicationForm from "../MedicationForm";
import "@test/matchMedia";

const mockFields = {
  id: 1,
  medicine: "MedicineName1",
  dosage: "dosage1",
  timesPerDay: "frequency1",
  instruction: "instruction1",
};

const mockRemoveMedicationInput = jest.fn();

describe("MedicationForm", () => {
  it("Snapshot test", () => {
    const { asFragment } = render(
      <MedicationForm
        fields={mockFields}
        removeMedicationInput={mockRemoveMedicationInput}
      />,
    );
    expect(screen.getByText("Medication")).toBeInTheDocument();
    expect(screen.getByLabelText("Medicine Name")).toBeInTheDocument();
    expect(screen.getByLabelText("Dosage")).toBeInTheDocument();
    expect(screen.getByLabelText("Frequency")).toBeInTheDocument();
    expect(screen.getByLabelText("Instruction")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("calls removeMedicationInput when remove field is clicked", async () => {
    render(
      <MedicationForm
        fields={mockFields}
        removeMedicationInput={mockRemoveMedicationInput}
        removeable={true}
      />,
    );
    await waitFor(() => {
      const removeFieldBtn = screen.getByTestId("remove-field");
      expect(removeFieldBtn).toBeInTheDocument();
      fireEvent.click(removeFieldBtn);
      expect(mockRemoveMedicationInput).toHaveBeenCalledWith(mockFields.id);
    });
  });
});
