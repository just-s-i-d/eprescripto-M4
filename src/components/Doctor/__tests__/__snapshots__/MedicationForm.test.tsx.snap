// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedicationForm Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="mt-4 mb-2 flex justify-between"
  >
    <h2
      class="text-xl"
    >
      Medication
    </h2>
  </div>
  <div
    class="w-full grid sm:grid-cols-2 md:grid-cols-4 gap-x-5 gap-y-3"
  >
    <div
      class="ant-form-item w-full css-dev-only-do-not-override-1adbn6x"
    >
      <div
        class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
      >
        <div
          class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
        >
          <label
            class="ant-form-item-required"
            for="MedicineName1"
            title="Medicine Name"
          >
            Medicine Name
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                aria-required="true"
                class="ant-input css-dev-only-do-not-override-1adbn6x"
                id="MedicineName1"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item w-full css-dev-only-do-not-override-1adbn6x"
    >
      <div
        class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
      >
        <div
          class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
        >
          <label
            class="ant-form-item-required"
            for="dosage1"
            title="Dosage"
          >
            Dosage
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                aria-required="true"
                class="ant-input css-dev-only-do-not-override-1adbn6x"
                id="dosage1"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item w-full css-dev-only-do-not-override-1adbn6x"
    >
      <div
        class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
      >
        <div
          class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
        >
          <label
            class="ant-form-item-required"
            for="frequency1"
            title="Frequency"
          >
            Frequency
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                aria-required="true"
                class="ant-select ant-select-in-form-item css-dev-only-do-not-override-1adbn6x ant-select-single ant-select-show-arrow"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-controls="frequency1_list"
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-owns="frequency1_list"
                      aria-required="true"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="frequency1"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-placeholder"
                  />
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex gap-6"
    >
      <div
        class="ant-form-item w-full css-dev-only-do-not-override-1adbn6x"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
        >
          <div
            class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
          >
            <label
              class="ant-form-item-required"
              for="instruction1"
              title="Instruction"
            >
              Instruction
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  aria-required="true"
                  class="ant-select ant-select-in-form-item css-dev-only-do-not-override-1adbn6x ant-select-single ant-select-show-arrow"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-controls="instruction1_list"
                        aria-expanded="false"
                        aria-haspopup="listbox"
                        aria-owns="instruction1_list"
                        aria-required="true"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="instruction1"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-placeholder"
                    />
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
